src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs

```diff
@@ -279,14 +320,23 @@ await foreach (var baseRecordField in baseRecord.GetFieldsAsync(cancellationToke
 
                     if (!compare1.Equals(compare2))
                     {
-                        ret = false;
-                        break;
+                        return false;
                     }
                 }
                 else if (baseRecordField.Value is RecordValue baseRecordValue && currentFieldValue is RecordValue currentRecordValue)
                 {
                     ret = await MatchesAsync(currentRecordValue, baseRecordValue, cancellationToken).ConfigureAwait(false);
                 }
+                else if (baseRecordField.Value is TableValue baseTableValue && currentFieldValue is TableValue currentTableValue)
+                {
+                    // Let's not scan throw entire table, but just check if the row count is the same.
+                    ret = baseTableValue.Count() == currentTableValue.Count();

```

Comment:
this impacts language semantics - double check?
Alternative - another way to compare? Or we just ignore tables completely?

Reply:
After verifying with DV, we don't need this change